### PR TITLE
Register items in the homepage's SiteSummaryPanel via hooks

### PR DESCRIPTION
**Pull request by gasman**
Currently the 'site summary' panel on the homepage (which shows the number of pages / images / docs on the site) relies on hard-coded references to wagtailimages and wagtaildocs within wagtailadmin. This makes it impossible to swap out those apps.

This PR adds a new hook `construct_homepage_summary_items` to allow wagtailimages and wagtaildocs to register these summary items themselves.

### CODE REVIEW

File: wagtail/wagtaildocs/wagtail_hooks.py [hunk 1]
```diff
@@ -59,3 +61,17 @@ def register_permissions():
...
+
+class DocumentsSummaryItem(SummaryItem):
+    order = 300
+    template = 'wagtaildocs/homepage/site_summary_documents.html'
+
+    def get_context(self):
+        return {
+            'total_docs': Document.objects.count(),
+        }
+
```
Comment (kaedroho) [inline wagtail/wagtaildocs/wagtail_hooks.py]: Missing blank line

**Comment by kaedroho** [comment]
Looks good to me

File: wagtail/wagtaildocs/wagtail_hooks.py (hunk 1 above)
Comment (gasman) [inline wagtail/wagtaildocs/wagtail_hooks.py]: Isn't it better to keep it visually grouped with the hook definition, to indicate that they're dealing with the same thing?

Comment (kaedroho) [inline wagtail/wagtaildocs/wagtail_hooks.py]: Yep, that's true. Ignore this

**Pull request merged**